### PR TITLE
ANSI-352: Do not use DH cipher suite

### PR DESCRIPTION
**Pull request by stan-confluent**
# Description

Fixes # ANSI-352

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:
Ran ansible for 1 zk, 1 kafka and 1 SR node on AWS rhel 8.2 ami without the change (got dh key too small error on SR healthcheck) and with the change (no error).
I'll be testing this further on a bigger environment.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: kerberos_kdc.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+all:
```
Comment (domenicbove) [inline kerberos_kdc.yml:1]: I dont think this file was meant to be included

File: kerberos.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- name: Provision KDC Server and Create Principals
```
Comment (domenicbove) [inline kerberos.yml:1]: same with this one

**Comment by domenicbove** [comment]
This is awesome! Thanks for making the PR, the question is should those properties be conditionally added only for rhel8 or should they be added for all os's.

Also we could take the stance that supporting rhel8 will come out in 6.0, in which case you should make your PR into the master branch. Alternatively we call this a bug in 55 and the PR is where it should be

And the final thing that we should add as part of this PR is a molecule scenario that tests rhel8 to validate it all. I can help with that.

File: kerberos_kdc.yml (hunk 1 above)
Comment (stan-confluent) [reply kerberos_kdc.yml:1]: yeah, you're right :)

File: kerberos.yml (hunk 2 above)
Comment (stan-confluent) [reply kerberos.yml:1]: fixed :)

**Comment by stan-confluent** [comment]
I think we should consult with other folks (security, release and product probably) on which versions should support it. My gut feeling is we want to avoid known vulnerabilities for any OS and version, not just rhel8. And I think rhel8 support is planned starting with 5.5 (probably 5.5.1?). OTOH one can argue that this should probably be solved in the code of the product itself, if this setting becomes semi-mandatory.
cc @sdanduConf @andrewegel

**Comment by JumaX** [comment]
@stan-confluent @domenicbove  Really great work here.  We don't officially support rhel 8 at this time, I think we should confirm if we will with 6.0 or not, in which case we can merge this into master for that release.

**Comment by Fobhep** [comment]
@stan-confluent Thank you for this fix! I have the same problem but with connect. Can you confirm your fix works on the health-check there? For me it sadly does not.

**Comment by JumaX** [comment]
@Fobhep Nothing that can be made public at this time unfortunately.

**Comment by Fobhep** [comment]
> @Fobhep Nothing that can be made public at this time unfortunately.

I see - thanks anyways :)

**Comment by domenicbove** [comment]
Closing because added in this pr: https://github.com/confluentinc/cp-ansible/pull/307

Will be released with 6.0

@Fobhep for the time being, you can either run with the `update-crypto-policies --set LEGACY && reboot` command, or try using the master branch, which has the ciphers

**Comment by stan-confluent** [comment]
@domenicbove & @Fobhep I'm probably stating the obvious, but LEGACY policies are probably less secure than current ones, so while it probably is a good way to get unblocked, I don't think it should be a long-term solution

**Comment by domenicbove** [comment]
Yes, agreed @stan-confluent the legacy policies is not a real solution, thats why the 60 branch will have this fix in it. Also I found an even better solution here: https://github.com/confluentinc/cp-ansible/pull/310

**Pull request closed without merging**